build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/ElasticsearchJavaModulePathPluginFuncTest.groovy

```diff
@@ -0,0 +1,202 @@
+        assertCompileClasspath([], normalized(result.output))
+        file('build/classes/java/main/module-info.class').exists()
+        assertModuleInfo(file('build/classes/java/main/module-info.class'), 'rootModule', [JAVA_BASE_MODULE])
+    }
+
+    def "module project with module dependencies"() {
+        given:
+        file('some-other-lib/src/main/java/module-info.java') << """
+        module someOtherLibModule {
+        }
+        """
+        file('some-lib/src/main/java/module-info.java') << """
+        module someLibModule {
+            requires someOtherLibModule;
+        }
+        """
+        file('src/main/java/module-info.java') << """
+        module rootModule {
+            requires someLibModule;
+        }
+        """
+        when:
+        def result = gradleRunner('compileJava').build()
+        then:
+        result.task(":compileJava").outcome == TaskOutcome.SUCCESS
+
+        assertModulePathClasspath(['./some-lib/build/classes/java/main', './some-other-lib/build/classes/java/main'], normalized(result.output))
+        assertCompileClasspath([], normalized(result.output))
+        file('build/classes/java/main/module-info.class').exists()
+        file('some-lib/build/classes/java/main/module-info.class').exists()
+        file('some-other-lib/build/classes/java/main/module-info.class').exists()
+        assertModuleInfo(file('build/classes/java/main/module-info.class'), 'rootModule', [JAVA_BASE_MODULE, 'someLibModule:'+ES_VERSION])
+    }
+
+    def "module project with transitive module dependency"() {
+        given:
+        file('src/main/java/module-info.java') << """
+        module rootModule {
+        }
+        """
+        file('some-other-lib/src/main/java/module-info.java') << """
+        module someOtherLibModule {
+        }
+        """
+
+        when:
+        def result = gradleRunner('compileJava').build()
+        then:
+        result.task(":compileJava").outcome == TaskOutcome.SUCCESS
+
+        assertModulePathClasspath(['./some-lib/build/classes/java/main', './some-other-lib/build/classes/java/main'], normalized(result.output))
+        assertCompileClasspath([], normalized(result.output))
+        file('build/classes/java/main/module-info.class').exists()
+        assertModuleInfo(file('build/classes/java/main/module-info.class'), 'rootModule', [JAVA_BASE_MODULE])
+    }
+
+    private def assertModulePathClasspath(List<String> expectedEntries, String output) {
+        def allArgs = output.find(/(?<=COMPILE_JAVA_COMPILER_ARGS ).*\n/).trim()
+        if(allArgs.isEmpty()) {
+            assert expectedEntries.size() == 0
+        } else {
+            def modulePathEntries = OperatingSystem.current.isWindows() ?
+                allArgs.find(/(?<=.*--module-path=)[^;]*(?=;)?/) :
+                allArgs.find(/(?<=.*--module-path=).*/).minus(";--module-version=${ES_VERSION}")
```

Comment:
sorry I think this needs to be reverted. meaning

```
 modulePathEntries = OperatingSystem.current.isWindows() ?
                 allArgs.find(/(?<=.*--module-path=).*/).minus(";--module-version=${ES_VERSION}") :
                 allArgs.find(/(?<=.*--module-path=)[^;]*(?=;)?/)
```

Reply:
D'oh! fixing